features/FxA-106-signin-unblock/README.md

```diff
@@ -0,0 +1,285 @@
+# Signin unblock
+
+## Problem Summary
+
+When our fraud-and-abuse system generates a false positive and blocks a legitimate user request, it currently results in a dead-end error message and a bad user experience. This makes us hesitant to increase the sensitivity of our security rules.
+
+When our fraud-and-abuse system generates a false negative and allows a malicious login attempt with a guessed or breached password, we have to rely on the sign-in confirmation feature as additional security for the user's sync data. The attacker still learns whether their login attempt succeeded, meaning they still learn whether they have the correct account password and can still access other FxA relying services.
+
+With this feature, we aim to lower the impact of false positives when blocking login attempts, by providing clearer messaging to the user about what's happening and why, and by allowing legitimate users to unblock their requests by proving ownership of the account. These measures will give us more confidence to tighten the security rules and thus reduce the rate of false negatives, improving overall security of the system.
+
+## Outcomes
+
+Users whose login attempts are blocked by our fraud-and-abuse system will be able to unblock themselves by doing an email verification loop to prove ownership of the account. As a result we will be able to:
+
+* Implement stricter rules to detect fraudulent access, and simultaneously
+* Decrease the number of legitimate login flows abandoned due to security measures.
+
+## Hypothesis
+
+If we send emails that allow users to unblock themselves after being flagged as suspicious by security measures, then we will increase the completion rate of legitimate login attempts without allowing significantly more fraudulent logins, because users will be able to immediately continue without waiting for a time period to elapse.
+
+We will know this to be true when we see an increase in the completion rate of login attempts that triggered a security measures, as a percentage of all login attempts
+that trigger a security measure.
+Metrics
+
+As a high-level success metric, we want to know the following:
+
+> Of all login attempts that were blocked a security measure, what percentage are subsequently completed successfully?
+
+We expect the starting value of this metric to be close to zero, since it is not obvious to users whether and how they can complete the flow after being blocked by a security measure. They currently can, in some cases, by waiting for a rate-limiting ban to expire, but this is not explained in the UI and users are unlikely to do so in practice.
+
+TODO: mock graph here
```

Comment:
A quick mock: ![20161018_114258](https://cloud.githubusercontent.com/assets/34695/19460710/e5ac2fec-9529-11e6-9330-7c4a01efad09.jpg)

If we're using flow events, will this graph be:

* 100 * (number of distinct flows containing `account.signin.sentUnblockCode` / number of distinct flows containing `account.signin.block`)
* 100 * (number of distinct flows containing `account.signin.confirmedUnblockCode` / number of distinct flows containing `account.signin.block`)

Or similar?